pkg/machine/qemu/machine.go

```diff
@@ -615,23 +616,57 @@ func (v *MachineVM) Start(name string, opts machine.StartOptions) error {
 		}
 	}
 
-	// If the qemusocketpath exists and the vm is off/down, we should rm
-	// it before the dial as to avoid a segv
-	if err := v.QMPMonitor.Address.Delete(); err != nil {
+	vlanSocket, err := machineSocket(v.Name, "vlan", "")
+	if err != nil {
 		return err
 	}
+	isFdVlanVM := false
+	for _, c := range v.CmdLine {
+		if c == fdVlanNetdev() {
+			isFdVlanVM = true
+		}
+	}
 
-	qemuSocketConn, err = machine.DialSocketWithBackoffs(maxBackoffs, defaultBackoff, v.QMPMonitor.Address.Path)
+	forwardSock, forwardState, forwarderProcess, err := v.startHostNetworking(vlanSocket)
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to start host networking: %q", err)
 	}
-	defer qemuSocketConn.Close()
+	defer func() {
+		_ = vlanSocket.Delete()
```

Comment:
I 100% agree here. I replicated the old behavior, where my understanding was the goal of reusing qmp socket path and then deleting is was to get some "mode security" by not allowing to connect to that socket after the connection is established. It's pretty wrong to delete the resource, which actually was allocated by another process. If the original design decision was made only by the fact that "we already have some path prepared - let's use it and just free it afterwards", then it is fine by me to change it to a less surprising implementation. Will fix this.

Reply:
I removed the part and it uncovered another bug with gvproxy being terminated with SIGKILL would keep the socket in the filesystem. So, I have to change it to SIGTERM. I also added a check to startup that the file is not present or is successfully cleared before starting gvproxy. For Windows it will still be WM_QUIT message, I'm yet to run tests on a Windows box if this works correctly, but this can be cleared in a follow up. Here is a diff for review:

```diff
diff --git a/pkg/machine/qemu/machine.go b/pkg/machine/qemu/machine.go
index b80e54509..49e46737e 100644
--- a/pkg/machine/qemu/machine.go
+++ b/pkg/machine/qemu/machine.go
@@ -536,6 +536,10 @@ func (v *MachineVM) Start(name string, opts machine.StartOptions) error {
 	if err != nil {
 		return err
 	}
+	err = vlanSocket.Delete()
+	if err != nil {
+		return err
+	}
 	isFdVlanVM := false
 	for _, c := range v.CmdLine {
 		if c == fdVlanNetdev() {
@@ -547,9 +551,6 @@ func (v *MachineVM) Start(name string, opts machine.StartOptions) error {
 	if err != nil {
 		return fmt.Errorf("unable to start host networking: %q", err)
 	}
-	defer func() {
-		_ = vlanSocket.Delete()
-	}()
 
 	if isFdVlanVM {
 		qemuSocketConn, err = machine.DialSocketWithBackoffs(maxBackoffs, defaultBackoff, vlanSocket.GetPath())
@@ -805,7 +806,7 @@ func (v *MachineVM) ProxyPID() (int, error) {
 // cleanupVMProxyProcess kills the proxy process
 func (v *MachineVM) cleanupVMProxyProcess(proxyPid int) error {
 	// Kill the process
-	if err := killProcess(proxyPid); err != nil {
+	if err := killProcess(proxyPid, false); err != nil {
 		return err
 	}
 	if err := v.PidFilePath.Delete(); err != nil {
@@ -853,7 +854,7 @@ func (v *MachineVM) Stop(_ string, _ machine.StopOptions) error {
 		return stopErr
 	}
 
-	if err := killProcess(qemuPid); err != nil {
+	if err := killProcess(qemuPid, true); err != nil {
 		if stopErr == nil {
 			return err
 		}
@@ -1220,7 +1221,7 @@ func (v *MachineVM) startHostNetworking(vlanSocket *define.VMFile) (string, mach
 	}
 
 	cmd := gvproxy.NewGvproxyCommand()
-	cmd.AddQemuSocket(fmt.Sprintf("unix://%s", strings.ReplaceAll(vlanSocket.GetPath(), "\\", "/")))
+	cmd.AddQemuSocket(fmt.Sprintf("unix://%s", filepath.ToSlash(vlanSocket.GetPath())))
 	cmd.PidFile = v.PidFilePath.GetPath()
 	cmd.SSHPort = v.Port
 
diff --git a/pkg/machine/qemu/machine_unix.go b/pkg/machine/qemu/machine_unix.go
index 385611117..ef47b94df 100644
--- a/pkg/machine/qemu/machine_unix.go
+++ b/pkg/machine/qemu/machine_unix.go
@@ -34,8 +34,12 @@ func pingProcess(pid int) (int, error) {
 	return pid, nil
 }
 
-func killProcess(pid int) error {
-	return unix.Kill(pid, unix.SIGKILL)
+func killProcess(pid int, force bool) error {
+	if force {
+		return unix.Kill(pid, unix.SIGKILL)
+	} else {
+		return unix.Kill(pid, unix.SIGTERM)
+	}
 }
 
 func checkProcessStatus(processHint string, pid int, stderrBuf *bytes.Buffer) error {
diff --git a/pkg/machine/qemu/machine_windows.go b/pkg/machine/qemu/machine_windows.go
index 9692b8eaa..8c5da41cb 100644
--- a/pkg/machine/qemu/machine_windows.go
+++ b/pkg/machine/qemu/machine_windows.go
@@ -26,7 +26,7 @@ func pingProcess(pid int) (int, error) {
 	return pid, nil
 }
 
-func killProcess(pid int) error {
+func killProcess(pid int, force bool) error {
 	machine.SendQuit(uint32(pid))
 	return nil
 }
```

Reply:
That looks good! BTW I have a consolidated win quit lib that gvproxy is now using here:https://github.com/containers/winquit

See the client example which also has a call that will attempt QUIT then fall back to kill after a timeout.
It's not critical of this PR just a mention.

Reply:
sure yeah not necessary just mentioning